BookShelfBusinessLogic/BookView.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace BookShelfBusinessLogic
+{
+    /// <summary>
+    /// Represents an entity for user-friendly representation of specific book
+    /// </summary>
+    public class BookView
+    {
+        /// <summary>
+        /// Initializes new instance of BookView class
+        /// </summary>
+        /// <param name="book"> Base book model </param>
+        /// <param name="context"> Data context </param>
+        public BookView(Book book, LibraryContext context)
+        {
+            this.Id = book.Id;
+            this.Title = book.Title;
+            Authors = new List<string>();
+            Genres = new List<string>();
+
+            foreach (BookAuthor item in context.BookAuthor)
+            {
+                if (item.BookRefId == book.Id)
+                {
+                    Author author = context.Authors.First(a => a.Id == item.AuthorRefId);
+                    this.Authors.Add(author.Name);
+                }
+            }
+
+            foreach (BookGenre item in context.BookGenre)
+            {
+                if (item.BookRefId == book.Id)
+                {
+                    Genre genre = context.Genres.First(a => a.Id == item.GenreRefId);
```

Comment:
I'm interested in what code convention are saying about naming of parameters in the LINQ. From one side meaningful name it is always good, but it would make query so long. So here is a question.

Reply:
Ups, it should be g as genre here. But, yes, overall, it is questionable.

Reply:
Anton has an answer on this, so at this moment it's right.